Sources/NIOTransportServices/NIOTSConnectionChannel.swift

```diff
@@ -891,4 +862,30 @@ fileprivate extension ChannelState where ActiveSubstate == NIOTSConnectionChanne
         }
     }
 }
+
+@available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+extension NIOTSConnectionChannel {
+    internal struct SynchronousOptions: NIOSynchronousChannelOptions {
+        @usableFromInline
+        internal let channel: NIOTSConnectionChannel
+
+        fileprivate init(channel: NIOTSConnectionChannel) {
+            self.channel = channel
+        }
+
+        @inlinable
+        public func setOption<Option: ChannelOption>(_ option: Option, value: Option.Value) throws {
+            try self.channel.setOption0(option: option, value: value)
+        }
+
+        @inlinable
+        public func getOption<Option: ChannelOption>(_ option: Option) throws -> Option.Value {
+            return try self.channel.getOption0(option: option)
+        }
+    }
+
+    public final var syncOptions: NIOSynchronousChannelOptions? {
```

Comment:
I don't think this needs a `final` annotation, the class is already `final`.

Reply:
Oof, thanks, too much copy-pasta :(